app/common/services/loadingProgress.service.js

```diff
@@ -0,0 +1,52 @@
+module.exports = LoadingProgress;
+
+LoadingProgress.$inject = ['$rootScope', '$transitions'];
+function LoadingProgress($rootScope, $transitions) {
+    function watchTransitions() {
+        $transitions.onStart({}, function (transition) {
+            $rootScope.isLoading = true;
+            if (transition.from().name === 'list.data.edit') {
+                $rootScope.isSaving = true;
+            }
+        });
+
+        $transitions.onFinish({}, function (transition) {
```

Comment:
What happens if a transition finishes before the http requests? ie. should this check for http requests in progress first?

Reply:
Good point, yes, it should check that first...